native/core/src/models/status.rs

```diff
@@ -60,20 +71,60 @@ impl TryFrom<AnyStateEvent> for RoomStatus {
                 inner: ActerSupportedRoomStatusEvents::RoomName(inner.content.name.clone()),
                 meta,
             }),
-            AnyStateEvent::RoomMember(StateEvent::Original(inner)) => inner
-                .membership_change()
-                .try_into()
-                .map(|change| RoomStatus {
-                    inner: ActerSupportedRoomStatusEvents::MembershipChange(MembershipChange {
-                        change,
-                        user_id: inner.state_key.clone(),
-                        display_name: inner.content.displayname.clone(),
-                        avatar_url: inner.content.avatar_url.clone(),
-                        reason: inner.content.reason.clone(),
-                    }),
+            AnyStateEvent::RoomMember(StateEvent::Original(inner)) => {
+                let user_id = inner.state_key.clone();
+                let membership_change = inner.content.membership_change(
+                    inner.prev_content().map(|c| c.details()),
+                    &inner.sender,
+                    &user_id,
+                );
+                let inner_status = if let MChange::ProfileChanged {
+                    displayname_change,
+                    avatar_url_change,
+                } = membership_change
+                {
+                    ActerSupportedRoomStatusEvents::ProfileChange {
+                        user_id,
+                        display_name_change: displayname_change.map(|c| Change {
+                            new_val: c.new.map(ToOwned::to_owned),
+                            old_val: c.old.map(ToOwned::to_owned),
+                        }),
+                        avatar_url_change: avatar_url_change.map(|c| Change {
+                            new_val: c.new.map(ToOwned::to_owned),
+                            old_val: c.old.map(ToOwned::to_owned),
+                        }),
+                    }
+                } else {
+                    let change = match membership_change {
+                        MChange::None => "None",
```

Comment:
this if else, match should be one big match instead IMHO

Reply:
Appended room state event handler for core object via https://github.com/acterglobal/a3/pull/2751/commits/c007e3dc00a0dde7eeac7035b6e7a99d9ed51be1.

Reply:
I could extend `ActerSupportedRoomStatusEvents` and `ActivityContent` easily by using `OtherState`.